code/modules/art/paintings.dm

```diff
@@ -184,13 +252,30 @@
 			if("red")
 				return "#ff0000"
 		return P.colour
-	else if(istype(I, /obj/item/soap) || istype(I, /obj/item/reagent_containers/glass/rag))
+	else if(istype(painting_implement, /obj/item/soap) || istype(painting_implement, /obj/item/reagent_containers/glass/rag))
 		return canvas_color
 
+/// Generates medium description
+/obj/item/canvas/proc/get_paint_tool_medium(obj/item/painting_implement)
+	if(!painting_implement)
+		return
+	if(istype(painting_implement, /obj/item/paint_palette))
+		return "Oil on canvas"
+	else if(istype(painting_implement, /obj/item/toy/crayon/spraycan))
+		return "Spraycan on canvas"
+	else if(istype(painting_implement, /obj/item/toy/crayon))
+		return "Crayon on canvas"
+	else if(istype(painting_implement, /obj/item/pen))
+		return "Ink on canvas"
+	else
+		return
```

Comment:
either remove this or have it return "Unknown medium"

Reply:
This is to handle cleaning tools that reset to canvas but i'll make this more explicit.